src/transforms/projectionpursuit.jl

```diff
@@ -0,0 +1,208 @@
+end
+
+function pindex(transform, Z, α)
+  X = rscore(Z, α)
+  I = (3/2) * mean(X)^2
+  if transform.deg > 1
+    Pⱼ₋₂, Pⱼ₋₁ = ones(size(X)), X
+    for j = 2:transform.deg
+      Pⱼ₋₂, Pⱼ₋₁ = 
+        Pⱼ₋₁, (1/j) * ((2j-1) * X .* Pⱼ₋₁ - (j-1) * Pⱼ₋₂)
+      I += ((2j+1)/2) * (mean(Pⱼ₋₁))^2
+    end
+  end
+  I
+end
+
+function optimaxis(transform, Z)
+  _, q = size(Z)
+  Iₘₐₓ, iₘₐₓ = 0, 0
+  Id = Matrix(1.0I, q, q)
+  for i = 1:q
+    Ip = pindex(transform, Z, Id[:,i])
+    if Ip > Iₘₐₓ
+      Iₘₐₓ, iₘₐₓ = Ip, i
+    end
+  end
+  Id[:,iₘₐₓ]
+end
+
+function optimdiag(transform, Z, ᾱ)
+  l  = length(ᾱ)
+  I₀ = pindex(transform, Z, ᾱ)
+  Id = Matrix(1.0I, l, l)
+  for i in 1:l
+    eᵢ = Id[:,i]
+    f₊ = pindex(transform, Z, (1/√(2+2ᾱ[i])) * (ᾱ + eᵢ))
+    f₋ = pindex(transform, Z, (1/√(2-2ᾱ[i])) * (ᾱ - eᵢ)) * (ᾱ[i] != 1)
+    f = (f₊ > f₋) * f₊ + (f₊ ≤ f₋) * f₋ 
+    s = (f₊ > f₋) * 1 + (f₊ ≤ f₋) * (-1)
+    if f > pindex(transform, Z, ᾱ)
+      ᾱ = (1/√(2+2s*ᾱ[i])) * (ᾱ + s*eᵢ)
+    end
+  end
+  pindex(transform, Z, ᾱ) ≥ I₀ ? ᾱ : optimaxis(transform, Z)
+end
+
+function optimgen(transform, Z, α₀)
+  f(a) = -pindex(transform, Z, a)
+  op = optimize(f, α₀)
+  minimizer(op)
+end
+
+function optim(transform, Z)
+  a_axis = optimaxis(transform, Z)
+  a_diag = optimdiag(transform, Z, a_axis)
+  optimgen(transform, Z, a_diag)  
+end
+
+# transforms the cdf of a convex combination of r.v. to a standard normal cdf
+function rmstructure(transform, Z, α)
+  l = length(α)
+  U = qr(hcat(α, rand(l, l-1)))
+  while norm(diag(U.R)) < transform.tol
+    U = qr(hcat(α, rand(l, l-1)))
+  end
+  T = Z * U.Q
+  table = Tables.table(T)
+  newtable, colcache = apply(Quantile(), table)
+  T[:,1] = collect(newtable)[1]
+  T * U.Q', U.Q, colcache
+end
+
+# gives the array that will be used to stop the loop of rmstructure 
+# by computing the projection index on samples of the standard multivariate
+# Gaussian distribution
+function gausspmatrix(transform, N, q)
+  Imatrix = rand(transform.n, q)
+  d = MvNormal(zeros(q), I)
+  Q = Matrix(1.0I, q, q) 
+  for i = 1:transform.n
+    S = rand(d, N)'
+    for j = 1:q
+      Imatrix[i,j] = pindex(transform, S, Q[:,j])
+    end
+  end
+  Imatrix
+end
+
+# Computes the pindex of each loop dataset (with possibly modified structure)
+# along q random directions and compares with the array given by gaussianpmatrix.
+function stopcriteria(transform, Z, Imatrix)
+  k, q = size(Imatrix)
+  Q = Matrix(1.0I, q, q) 
+  switch = true
+  for i = 1:q
+    b = pindex(transform, Z, Q[:,i])
+    switch *= quantile(Imatrix[1:k-1,i], transform.perc) > b
+  end
+  !switch
+end
+
+#Generates the loop for the structure removal process
+function applyfeat(transform::ProjectionPursuit, table, prep) 
+  names = Tables.columnnames(table)
+
+  trans = Quantile() → EigenAnalysis(:VDV)
+  ttable, tcache = apply(trans, table)
+  Z = Tables.matrix(ttable)
+  N,q = size(Z)
+  Imatrix = gausspmatrix(transform, N, q) 
```

Comment:
Let's try to follow a clearer convention for these matrices with projections scores. In particular, if the matrix corresponds to the ideal Gaussian case, then it should probably have that in its name, for example: Igauss

Please try to review all matrices involving scores in the code to help the readers. There are too many variations of the variable name to keep track still.